### PR TITLE
MINOR: fixing JavaDocs and other cleanup

### PR DESCRIPTION
**Pull request by mjsax**
Thing IntelliJ highlighted...

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java [hunk 1]
```diff
@@ -1452,7 +1456,7 @@ public CloseOptions leaveGroup(final boolean leaveGroup) {
      * This will block until all threads have stopped.
      */
     public void close() {
-        close(Long.MAX_VALUE, false);
+        close(Optional.empty(), false);
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java:1459]: Changing this to `Optional` to unify the log message (cf other changes).

File: streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java [hunk 2]
```diff
@@ -539,9 +539,12 @@ private void handleStreamsUncaughtException(final Throwable throwable,
                 }
                 break;
             case SHUTDOWN_CLIENT:
-                log.error("Encountered the following exception during processing " +
-                        "and the registered exception handler opted to " + action + "." +
-                        " The streams client is going to shut down now. ", throwable);
+                log.error(String.format(
+                    "Encountered the following exception during processing and the registered exception handler" +
```
Comment (chia7712) [inline streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java:543]: Pardon me, why not using `{}` placeholder directly?

File: streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java [hunk 3]
```diff
@@ -1531,7 +1535,15 @@ private Thread shutdownHelper(final boolean error, final long timeoutMs, final b
         }, clientId + "-CloseThread");
     }
 
-    private boolean close(final long timeoutMs, final boolean leaveGroup) {
+    private boolean close(final Optional<Long> timeout, final boolean leaveGroup) {
+        final long timeoutMs;
+        if (timeout.isPresent()) {
+            timeoutMs = timeout.get();
+            log.debug("Stopping Streams client with timeoutMillis = {} ms.", timeoutMs);
```
Comment (chia7712) [inline streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java]: nit: there is "ms" already, so `timeoutMillis` could be simplified to `timeout`

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 4]
```diff
@@ -1309,7 +1309,11 @@ public static boolean getBoolean(final Map<String, Object> configs, final String
             } else if (value instanceof String) {
                 return Boolean.parseBoolean((String) value);
             } else {
-                log.warn("Invalid value (" + value + ") on internal configuration '" + key + "'. Please specify a true/false value.");
+                log.warn(String.format(
```
Comment (chia7712) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: maybe we use placeholder?
```java
                log.warn(
                    "Invalid value ({}) on internal configuration '{}}'. Please specify a true/false value.",
                    value,
                    key
                );
```

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 5]
```diff
@@ -1321,7 +1325,11 @@ public static long getLong(final Map<String, Object> configs, final String key,
             } else if (value instanceof String) {
                 return Long.parseLong((String) value);
             } else {
-                log.warn("Invalid value (" + value + ") on internal configuration '" + key + "'. Please specify a numeric value.");
+                log.warn(String.format(
```
Comment (chia7712) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: ditto

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 6]
```diff
@@ -1331,7 +1339,11 @@ public static String getString(final Map<String, Object> configs, final String k
             if (value instanceof String) {
                 return (String) value;
             } else {
-                log.warn("Invalid value (" + value + ") on internal configuration '" + key + "'. Please specify a String value.");
```
Comment (chia7712) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:1306]: ditto

**Review by chia7712 — COMMENTED** [review]
@mjsax thanks for this patch. overall LGTM

It seems `StreamConfig` prefers to use `String.format` rather than slf4j "parameterized messages". It is cool to have consistent code style, but using slf4j format is more simple and fast (https://www.slf4j.org/faq.html#logging_performance).

**Comment by chia7712** [comment]
@mjsax Could you please fix the conflicts?

**Review by chia7712 — APPROVED** [review]

**Pull request merged**